packages/integration-tests/tests/users.test.ts

```diff
@@ -0,0 +1,86 @@
+import { User } from '@logto/schemas';
+
+import { authedAdminApi } from '@/api';
+import { generatePassword, generateUsername } from '@/utils';
+
+const createUser = async () => {
+  const randomUsername = generateUsername();
+
+  return authedAdminApi
+    .post('users', {
+      json: {
+        username: randomUsername,
+        password: generatePassword(),
+        name: randomUsername,
+      },
+    })
+    .json<User>();
+};
+
+describe('admin console user management', () => {
+  it('should create user successfully', async () => {
+    const user = await createUser();
+
+    expect(user).toBeTruthy();
+  });
+
+  it('should get user list successfully', async () => {
+    await createUser();
+    const users = await authedAdminApi.get('users').json<User[]>();
+
+    expect(users.length).not.toBeLessThan(1);
+  });
+
+  it('should get user details successfully', async () => {
+    const user = await createUser();
+
+    const userDetails = await authedAdminApi.get(`users/${user.id}`).json<User>();
+
+    expect(userDetails).toBeTruthy();
+  });
+
+  it('should update userinfo successfully', async () => {
+    const user = await createUser();
+
+    const newUserData: Partial<User> = {
+      name: 'new name',
+      avatar: 'https://new.avatar.com/avatar.png',
+      customData: {
+        level: 1,
+      },
+      roleNames: ['admin'],
+    };
+
+    const updatedUser = await authedAdminApi
+      .patch(`users/${user.id}`, {
+        json: newUserData,
+      })
+      .json<User>();
+
+    expect(updatedUser).toMatchObject(newUserData);
```

Comment:
To contain? Response User type not guaranteed same as your Partial<User>

Reply:
`toMatchObject` is ok here?
<img width="788" alt="image" src="https://user-images.githubusercontent.com/10806653/180167543-1de97d17-3742-49ac-9460-6a7e1599c28f.png">
<img width="672" alt="image" src="https://user-images.githubusercontent.com/10806653/180167620-dace2f74-bad2-4a48-abc0-c7a12bf34b6e.png">

Reply:
o ok my bad sounds good.